src/test_report.py

```diff
@@ -125,7 +125,24 @@ def _create_test_report(self, root_node):
         )
         return content, subject
 
-    def run(self):
+    def _send_email(self, content, subject):
+        smtp = self._smtp_connect()
+        if smtp:
+            email_msg = self._create_email(
+                self._email_send_to, self._email_send_from,
+                subject, content
+            )
+            smtp.send_message(email_msg)
+            smtp.quit()
```

Comment:
What is the reason to keep this method as an internal API?
I don't see the point why it can't be public. Having said that, I think we should consider separating e-mail related logic and the report. I think we should have 2 separate classes for it.

Reply:
This service doesn't provide any API, it's only a command line tool.  It wasn't designed to be imported and reused.  This PR is a refactor and the code to send an email is being moved to a separate method, but that's all implementation detail.

The service needs to have an event loop so that every time a kernel revision has been tested a report is generated and sent.  So that needs to be done within the service's code.  The class here is only to provide the command line interface like in the `kci_*` tools.

We could rework this further by separating the logic that creates the email report (which probably could go into kernelci-core anyway) and the logic that sends emails, but I think that's beyond the scope of this PR.